service/history/timerQueueProcessor.go

```diff
@@ -540,7 +540,7 @@ Update_History_Loop:
 
 		// We apply the update to execution using optimistic concurrency.  If it fails due to a conflict than reload
 		// the history and try the operation again.
-		err := t.updateWorkflowExecution(context, msBuilder, scheduleNewDecision, timerTasks, nil)
+		err := t.updateWorkflowExecution(context, msBuilder, scheduleNewDecision, timerTasks, nil, []persistence.Task{})
```

Comment:
The internal updateWorkflowExecution does an append of decision schedule task, so empty slice. i removed this and exposed through flag createDeletionTask.